packages/flutter/lib/src/services/text_input.dart

```diff
@@ -633,6 +633,11 @@ abstract class TextInputClient {
 
   /// Updates the floating cursor position and state.
   void updateFloatingCursor(RawFloatingCursorPoint point);
+
+  /// Platform notified framework of closed connection.
+  ///
+  /// [TextInputClient] should cleanup its connection and finalize editing.
+  void connectionClosed();
```

Comment:
Sorry, I forgot to notice this before, but this looks like a breaking change. Subclasses/Implementors of TextInputClient will have to be changed. We should follow the breaking change process, maybe since up with Dan since he also has a breaking change in this area: https://github.com/flutter/flutter/pull/43959

Reply:
Thanks for noticing this. I synced with @dnfield I'll send the email for both of us now.